### PR TITLE
Add information about bytes downloaded

### PR DESCRIPTION
**Pull request by mykhailo-kuchma**
Add information about bytes downloaded to the prefetch status callback.
Add network statistics to HttpResponse
Add network statistics PrefetchStatus

Relates-To: OLPEDGE-2169

Signed-off-by: Mykhailo Kuchma <ext-mykhailo.kuchma@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/src/ApiResponseWithNetworkStats.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+#include <olp/core/client/ApiResponse.h>
+#include <olp/core/client/HttpResponse.h>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+template <typename Result, typename Error>
+class ApiResponseWithNetworkStats : public client::ApiResponse<Result, Error> {
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/ApiResponseWithNetworkStats.h:30]: How about `ExtendedApiResponse` and make it more generic with an additional template parameter instead of fixed to NetworkStatistics? Might be more flexible for the future.

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h [hunk 2]
```diff
@@ -39,7 +41,9 @@ namespace read {
  */
 class BlobApi {
  public:
-  using DataResponse = client::ApiResponse<model::Data, client::ApiError>;
+  using DataResponseWithNetworkStats =
+      ExtendedApiResponse<model::Data, client::ApiError,
+                          client::NetworkStatistics>;
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h:46]: Why did you change it? Just leave DataResponse, it doesn't matter if there is additional information inside.

File: olp-cpp-sdk-dataservice-read/src/repositories/DataRepository.h [hunk 3]
```diff
@@ -35,25 +37,27 @@ namespace repository {
 
 class DataRepository final {
  public:
+  using DataResponseWithNetworkStats = BlobApi::DataResponseWithNetworkStats;
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/repositories/DataRepository.h:40]: Same here, I would keep `DataResponse`.

File: olp-cpp-sdk-dataservice-read/src/ExtendedApiResponse.h [hunk 4]
```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
```
Comment (kzvieriev) [inline olp-cpp-sdk-dataservice-read/src/ExtendedApiResponse.h:2]: 2020

File: olp-cpp-sdk-dataservice-read/src/ExtendedApiResponseHelpers.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
```
Comment (kzvieriev) [inline olp-cpp-sdk-dataservice-read/src/ExtendedApiResponseHelpers.h:2]: 2020

File: olp-cpp-sdk-dataservice-read/src/ExtendedApiResponse.h [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  ExtendedApiResponse(Result result)
+      : client::ApiResponse<Result, Error>(std::move(result)), payload_{} {}
+
+  // Implicit constructor by desing, same as in ApiResponse
+  ExtendedApiResponse(const Error& error)
+      : client::ApiResponse<Result, Error>(error), payload_{} {}
+
+  ExtendedApiResponse(Result result, Payload payload)
+      : client::ApiResponse<Result, Error>(std::move(result)),
+        payload_(payload) {}
```
Comment (kzvieriev) [inline olp-cpp-sdk-dataservice-read/src/ExtendedApiResponse.h:43]:
```suggestion
        payload_(std::move(payload)) {}
```
same as for `result` above.

File: olp-cpp-sdk-dataservice-read/src/ExtendedApiResponse.h [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+  ExtendedApiResponse(const Error& error, Payload payload)
+      : client::ApiResponse<Result, Error>(error), payload_(payload) {}
```
Comment (kzvieriev) [inline olp-cpp-sdk-dataservice-read/src/ExtendedApiResponse.h:46]:
```suggestion
      : client::ApiResponse<Result, Error>(error), payload_(std::move(payload)) {}
```

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h [hunk 8]
```diff
@@ -23,7 +23,9 @@
 
 #include <olp/core/client/ApiError.h>
 #include <olp/core/client/ApiResponse.h>
+#include <olp/core/client/HttpResponse.h>
 #include <boost/optional.hpp>
+#include "../../ExtendedApiResponse.h"
```
Comment (kzvieriev) [inline olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h:28]:
```suggestion
#include "ExtendedApiResponse.h"
```
I believe, olp-cpp-sdk-dataservice-read/src folder added to include directories.

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp [hunk 9]
```diff
@@ -29,14 +29,14 @@
...
-                                       boost::optional<std::string> billing_tag,
-                                       boost::optional<std::string> range,
-                                       const client::CancellationContext& context) {
+namespace client = olp::client;
+
+BlobApi::DataResponse BlobApi::GetBlob(
+    const client::OlpClient& client, const std::string& layer_id,
+    const std::string& data_handle, boost::optional<std::string> billing_tag,
+    boost::optional<std::string> range,
+    const client::CancellationContext& context) {
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp:39]: I don't understand this back and forth formatting.

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h (hunk 8 above)
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h:28]: This seems very ugly. Maybe move `ExtendedApiResponse` to `generated/api/`?

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp (hunk 9 above)
Comment (mykhailo-kuchma) [reply olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp:39]: clang-format 8.0.0 was used previously

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h (hunk 8 above)
Comment (mykhailo-kuchma) [reply olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h]: Replaced with #include "ExtendedApiResponse.h"
